internals-js/src/argumentCompositionStrategies.ts

```diff
@@ -59,4 +59,41 @@ export const ARGUMENT_COMPOSITION_STRATEGIES = {
         return acc.concat(newValues);
       }, []),
   },
+  NULLABLE_AND: {
+    name: 'NULLABLE_AND',
+    isTypeSupported: supportFixedTypes((schema: Schema) => [schema.booleanType()]),
+    mergeValues: (values: (boolean | null | undefined)[]) => values.reduce((acc, next) => {
+      if (acc === null || acc === undefined) {
+        return next;
+      } else if (next === null || next === undefined) {
+        return acc;
+      } else {
+        return acc && next;
+      }
+    }, undefined),
+  },
+  NULLABLE_MAX: {
+    name: 'NULLABLE_MAX',
+    isTypeSupported: supportFixedTypes((schema: Schema) => [schema.intType(), new NonNullType(schema.intType())]),
+    mergeValues: (values: any[]) => values.reduce((a: any, b: any) => a !== undefined && b !== undefined ? Math.max(a, b) : a ?? b, undefined),
+  },
+  NULLABLE_UNION: {
+    name: 'NULLABLE_UNION',
+    isTypeSupported: (_: Schema, type: InputType) => ({ valid: isListType(type) }),
+    mergeValues: (values: any[]) => {
+      if (values.every((v) => v === undefined)) {
+        return undefined;
+      }
+
+      const combined = new Set();
```

Comment:
We've talked recently about preserving ordering of directives. I know this might be a bit of a lost cause, but should we consider trying to do a better job of preserving the original ordering as best we can?

Reply:
The JS `Set` should preserve insertion order, so I think the ordering of arguments should be the same after this concatenation.